.github/workflows/nightly-devnet-test.yml

```diff
@@ -0,0 +1,373 @@
+name: Devnet network tests
+on:
+  pull_request:
+    paths:
+      - '**.yml'
+  workflow_dispatch:
+  schedule:
+    - cron: "00 08 * * 1-5"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  # This is a workaround because the runner does not have perms to upload files anywhere else
+  CONTRACT_S3_BUCKET: s3://aztec-ci-artifacts
+jobs:
+  cli-wallet:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+
+      - name: Reset state
+        run: |
+          echo '{"accounts": [], "contracts": []}'> state.json
+          aws s3 cp ./state.json ${{ env.CONTRACT_S3_BUCKET }}/build-cache/devnet-nightly-tests-state.json
+
+      - name: Install aztec
+        run: |
+          set -eu -o pipefail
+          bash -i <(curl -s https://install.aztec.network)
+          echo "/home/runner/.aztec/bin" >> $GITHUB_PATH
+
+      - name: Specify aztec version
+        run: |
+          set -eu -o pipefail
+          aztec-up -v 0.81.0
+
+      - name: Retrieve state from s3
+        run: |
+          set -eu -o pipefail
+          aws s3 cp ${{ env.CONTRACT_S3_BUCKET }}/build-cache/devnet-nightly-tests-state.json ./state.json || echo '{"accounts": [], "contracts": []}'> state.json
+          cat ./state.json
+
+      - name: Restore state
+        run: |
+          # Read the entire accounts array and process each item
+          jq -c '.accounts[]' state.json | while read -r account; do
+            address=$(echo $account | jq -r '.address')
+            secretKey=$(echo $account | jq -r '.secretKey')
+
+            aztec-wallet create-account --secret-key $secretKey --register-only --node-url http://104.198.9.16:8080
+
+          done
+
+      - name: Add contracts
+        id: add-contracts
+        run: |
+          aztec-wallet import-test-accounts --node-url http://104.198.9.16:8080
+          # Register all existing token contracts
+          jq -c '.contracts | map(select(.type=="token"))[]' state.json | while read -r contract; do
+            token_address=$(echo "$contract" | jq -r '.address')
+
+            echo "Processing token contract at $token_address"
+            # We register the token contracts with the test account because the setup elects to mint with it
+            aztec-wallet register-contract "$token_address" Token -f accounts:test0 --node-url http://104.198.9.16:8080
```

Comment:
This all seems quite insane - why write large bash scripts in yaml? Both the bash and yaml become unreadable: the bash is mixed in the whole gh actions thing, and what each action does is so large that navigating the yaml is messy. We don't even get syntax highlighting on the bash! How is one supposed to run these scripts locally?